docs/DeveloperGuide.md

```diff
@@ -165,6 +165,33 @@ The time allocation mechanism is facilitated by `TimeAllocator`. It allocates ti
 
 * Work in progress hehe
 
+Implementation
+TimeAllocator takes in  TaskList and separates it into a fixedTaskList and a flexibleTaskList
+
+When runAllocate is executed, a ForecastFilter is created in order to separate the tasks based on their days in order to identify the day that they belong to.
+
+An ArrayList named dayLog is used to keep track of the times that the tasks take up during the day, it stores the taskNumber that occupies each of the hourly slots.
+
+Fixed tasks are placed into the dayLog first as they do not change. 
+
+The start time and sleep time is referenced together with the dayLog to identify the vacant slots.
+
+bestLog identifies the tasks from the flexibleTaskList that reduces the dead space in the logs based on changing the tasks available for getLog to insert into the logs.
```

Comment:
Suddenly there is a getLog that came out of nowhere. Would be good to explain it.